### PR TITLE
Fix #125, enable `hasattr` for SafeDict.update

### LINKED ISSUE #125: `SafeDict.update` needs `hasattr`
Short and simple RepyV2 test:

``` python
a = SafeDict({"1":2})
a.update({"3":4})
log(a, "\n")
```

Problem:

```
(...)
  "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/UserDict.py", line 188, in update

Unsafe call: ("Unsafe call 'hasattr' with args '({'3': 4}, 'iteritems')', kwargs '{}'",)
```

Fix:

``` diff
97a98
> UserDict.hasattr = hasattr
```

Other `dict` features likely need the same / a similar fix, and tests too!

### PR DESCRIPTION
**Pull request by aaaaalbert**
This PR fixes #125: It
* updates `safe.py` to allow `hasattr` for `UserDict` (and thus `SafeDict`), and
* provides a test for many of the methods of `safe.SafeDict`, and particularly checks that `SafeDict.update` works now.

Note: I suggest to *squash-merge* this PR if you think it's fine. The separate commits are not too interesting.

### CODE REVIEW

File: testsV2/ut_repyv2api_safedictests.r2py [hunk 1]
```diff
@@ -0,0 +1,179 @@
+"""
+Test interactions with SafeDict.
+TODO: Add tests for the view* and iter* methods too!
```
Comment (lukpueh) [inline testsV2/ut_repyv2api_safedictests.r2py]: AFAICS, `SafeDict` does not have `view*` methods

File: testsV2/ut_repyv2api_safedictests.r2py [hunk 2]
```diff
@@ -0,0 +1,179 @@
+"""
+Test interactions with SafeDict.
```
Comment (lukpueh) [inline testsV2/ut_repyv2api_safedictests.r2py]: Would you mind listing all the interactions, i.e. `SafeDict(UserDict.DictMixin)` methods here?

File: testsV2/ut_repyv2api_safedictests.r2py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+if another_key not in s20 or a_default not in s20.values():
+  log("Setdefault did not set the desired value", s20, "\n")
+
+
+########################
+# Test SafeDict update, see SeattleTestbed/repy_v2#125
+
+# From another dict
+s3 = SafeDict({"1":2, "3":4})
```
Comment (lukpueh) [inline testsV2/ut_repyv2api_safedictests.r2py]: Missing whitespaces :)

File: testsV2/ut_repyv2api_safedictests.r2py (hunk 2 above)
Comment (aaaaalbert) [reply testsV2/ut_repyv2api_safedictests.r2py]: I'll list everything I test, which is mostly congruent with what `dict` allows, save for the (outcommented) creation methods from kwargs etc.

The `SafeDict` methods aren't quite the same thing, and we don't / can't (in Repy) access the double-underscore methods. `in`, `del` etc. are tested implicitly.

File: testsV2/ut_repyv2api_safedictests.r2py (hunk 3 above)
Comment (aaaaalbert) [reply testsV2/ut_repyv2api_safedictests.r2py]: Well spotted!

**Pull request merged**